docs/api.md

```diff
@@ -7,102 +7,167 @@ method arguments and return values.
-| Difficulty | Optional | `difficulty` | 0-100 difficulty rating |
-| Appeared In | Optional | `appeared_in` | String name of original competition |
-| Input Files | __Required__ | `in_file` | Zipped (.zip) directory of all input files |
-| Output Files | __Required__ | `out_file` | Zipped (.zip) directory of all output files |
-| Solution File | __Required__ | `sol_file` | Solution program (not zipped) |
 
-For __Sample Cases__, must be a string JSON array of mappings `input` and `output`.
-For example,
+From here, any user can perform the following actions:
+
+1. [Get Data on all Problems](#get-data-on-all-problems)
+2. [Get Data on a Specific Problem](#get-data-on-a-problem)
+
+Those logged in as an administrator can perform additional actions:
+
+3. [Create a New Problem](#create-a-new-problem)
+4. [Edit an Existing Problem](#edit-an-existing-problem)
+5. [Delete a Problem](#delete-a-problem)
+
+
+## Problem Management Objects
+
+### The Problem Object
+
+| Name | Type | Description |
+| --- | --- | --- |
+|`added`|`int`|The UTC timestamp indicating when this problem was uploaded|
+|`appeared`|`String`|The contest this problem originally appeared in|
+|`comp_release`|`int`|The competition `cid` for the competition this was used in|
+|`description`|`String`|The problem's description|
+|`difficulty`|`int`|The difficulty of the problem from 0 (easiest) to 100 (hardest)|
+|`input_desc`|`String`|A description of the problem's input|
+|`name`|`String`|The name of the problem|
+|`output_desc`|`String`|A description of the problem's output|
+|`pid`|`int`|The problem's unique id.|
+|`sample_cases`|`Sample Case[]`|A list of samples cases, as described below|
+|`shortname`|`String`|A unique string with no spaces used to identify the problem|
+
+The `sample_cases` field consists of an array of [Sample Case](#the-sample-case-object) objects.
+
+
+__Example:__
 
 ```json
-[
-  {
+{
+  "added": 1426310855,
+  "appeared": "2014 Mid-Central",
+  "comp_release": 6,
+  "description": null,
+  "difficulty": "67",
+  "input_desc": null,
+  "name": "(More) Multiplication",
+  "output_desc": null,
+  "pid": 34,
+  "sample_cases": [],
+  "shortname": "multiplication"
+}
+```
+
+The above problem object represents a problem that was added to the website March 14, 2015, at 12:30 AM, that
+originally appeared in the 2014 Mid-Central ICPC. It was first used in an AUACM competition with id 6. The
+problem has no description, input description, output description, or sample cases. It is named (More) Multiplication,
+with a problem id of 34, and is also uniquely identified by the string "multiplication". (More) Multiplication
+has a difficulty of 67.
+
+### The Sample Case Object
+
+| Name | Type | Description |
+| --- | --- | --- |
+|`input`|`String`|The input fed to standard in for a test run|
+|`output`|`String`|The output that would be produced to standard out by a correct program for that test run|
+
+__Example:__
+
+```json
+{
```

Comment:
Minor thing, but if you can find a sample with more data, it may be better. Totally up to you.